src/elf.rs

```diff
@@ -260,13 +261,26 @@ pub(crate) enum Section {
     Borrowed(usize, Range<usize>),
 }
 
+/// Defines a set of capabilities of an executable
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum ExecutableCapabilities {
```

Comment:
Wouldn't it be better if this was 

```
#[deive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd)]
enum SbpfVersion {
	V1 = 1,
	V2 = 2,
	V3 = 3,
}

impl SbpfVersion {
	fn has_static_syscalls(&self) -> bool {
		self >= SbpfVersion::V2
	}

	fn allows_elf_virtual_addresses(&self) -> bool { ... }

	...
}
```

Then in the call sites it would be obvious from the `has_*` name what's being
checked for, and we wouldn't have to go and edit call sites if we flip things on
and off across versions.